### Task 1: Import pandas

In [4]:
import pandas as pd
print(pd.__version__)

2.3.3


### Task 2: Load the Customers Dataset

In [5]:
customers = pd.read_csv('data/Customer_Dataset.csv ')
customers

,age,income,visits_per_month
0,30.483571,33893.885591,9.295377
1,35.615149,33126.773002,7.531726
2,35.896064,41139.477833,7.061051
3,30.712800,31292.658458,7.068540
4,29.209811,19693.758043,4.550164
...,...,...,...
295,67.926951,103758.922780,2.640708
296,59.134584,111630.553709,3.820482
297,58.550918,111000.120344,4.169296
298,64.675113,104730.648314,2.832882


### Task 3: Preview the Customers Data

In [6]:
customers.head(6)


,age,income,visits_per_month
0,30.483571,33893.885591,9.295377
1,35.615149,33126.773002,7.531726
2,35.896064,41139.477833,7.061051
3,30.712800,31292.658458,7.068540
4,29.209811,19693.758043,4.550164
5,25.188562,26897.351037,8.628495


In [7]:
customers.tail(4)


,age,income,visits_per_month
296,59.134584,111630.553709,3.820482
297,58.550918,111000.120344,4.169296
298,64.675113,104730.648314,2.832882
299,56.026996,113097.634498,2.183064


In [8]:
customers.sample(5)

,age,income,visits_per_month
221,51.196027,84307.313260,3.106430
189,33.951482,42550.496948,4.472730
295,67.926951,103758.922780,2.640708
244,57.857769,69461.245936,4.029156
168,36.096145,59154.495395,4.111409


Q: do random rows look different every time its run?
A: Yes, because it picks random samples

### Task 4: Load and inspect the cars dataset

In [15]:
cars = pd.read_csv("data/cars.csv")
print(customers.shape) 
print(customers.columns)

(300, 3)
Index(['age', 'income', 'visits_per_month'], dtype='object')


### Q: How many rows
### A: 300 
### How many columns
### A: 3 
### Column names:
### A: age, income, visits_per_month

### Task 5: Check data types in cars


In [16]:
customers.dtypes

age                 float64
income              float64
visits_per_month    float64
dtype: object

Q: which columns hold whole numbers:
A: none
Q: Which columns hold decimal numbers:
A: visits, age, income
Q: which columns hold text or mixed valuse:
A: none

### Task 6: Load and inspect the titanic Dataset

In [24]:
Titanic = pd.read_csv('data/Titanic Dataset.csv')
Titanic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   survived     891 non-null    int64  
 1   pclass       891 non-null    int64  
 2   sex          891 non-null    object 
 3   age          714 non-null    float64
 4   sibsp        891 non-null    int64  
 5   parch        891 non-null    int64  
 6   fare         891 non-null    float64
 7   embarked     889 non-null    object 
 8   class        891 non-null    object 
 9   who          891 non-null    object 
 10  adult_male   891 non-null    bool   
 11  deck         203 non-null    object 
 12  embark_town  889 non-null    object 
 13  alive        891 non-null    object 
 14  alone        891 non-null    bool   
dtypes: bool(2), float64(2), int64(4), object(7)
memory usage: 92.4+ KB


How many rows and columns:
A: 891 rows and 15 columns
Which column has the most missing values
A: deck
How do you know a value is missing
A: there are 891 entries and the non-null count provides the values for each column that aren't null

### Task 7 Summary statistics on cars

In [39]:
cars.describe()


,mpg,cylinders,displacement,weight,acceleration,model year,origin
count,398.000000,398.000000,398.000000,398.000000,398.000000,398.000000,398.000000
mean,23.514573,5.454774,193.425879,2970.424623,15.568090,76.010050,1.572864
std,7.815984,1.701004,104.269838,846.841774,2.757689,3.697627,0.802055
min,9.000000,3.000000,68.000000,1613.000000,8.000000,70.000000,1.000000
25%,17.500000,4.000000,104.250000,2223.750000,13.825000,73.000000,1.000000
50%,23.000000,4.000000,148.500000,2803.500000,15.500000,76.000000,1.000000
75%,29.000000,8.000000,262.000000,3608.000000,17.175000,79.000000,2.000000
max,46.600000,8.000000,455.000000,5140.000000,24.800000,82.000000,3.000000


What is the mean mpg?
A: 23.514573
What is the max HP
A: skipping for now
What is the median weight in this dataset?
A: 2803.5

### Task 8: Summary statisitics on one column

In [46]:
print("mean", Titanic['fare'].mean())
print("median", Titanic['fare'].median())
print("standard dev", Titanic['fare'].std())
print("min", Titanic['fare'].min())
print("max", Titanic['fare'].max())


mean 32.204207968574636
median 14.4542
standard dev 49.693428597180905
min 0.0
max 512.3292


### Task 9 build a DF from scratch

In [50]:
Store = {"product":["apples", "oranges", "grapes", "bananas"], "price":[1.99, 2.00, 3.00, 4.00], "in_stock":[True, True, False, False]}
DF = pd.DataFrame(Store)
DF.dtypes

product      object
price       float64
in_stock       bool
dtype: object

### Task 10 Discussion

1.  Q: diff between shape and info
    A: shape just gives just rows and columns, info gives everything
2.  Q: What is the diff between head and sample, when would sample be more useful?
    A: head gives the first five columns and sample is random rows. When you want random data sampling.
3.  Q: Titanic deck column had a lot missing values. Why would it be missing?
    A: Info may not be as easily accesible through articles provided for datasets. For Titanic, deck missing values correlates with "alive" column.

In [60]:
Titanic.sample(10)

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
605,0,3,male,36.0,1,0,15.5500,S,Third,man,True,NaN,Southampton,no,False
837,0,3,male,NaN,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True
737,1,1,male,35.0,0,0,512.3292,C,First,man,True,B,Cherbourg,yes,True
56,1,2,female,21.0,0,0,10.5000,S,Second,woman,False,NaN,Southampton,yes,True
478,0,3,male,22.0,0,0,7.5208,S,Third,man,True,NaN,Southampton,no,True
573,1,3,female,NaN,0,0,7.7500,Q,Third,woman,False,NaN,Queenstown,yes,True
457,1,1,female,NaN,1,0,51.8625,S,First,woman,False,D,Southampton,yes,False
62,0,1,male,45.0,1,0,83.4750,S,First,man,True,C,Southampton,no,False
838,1,3,male,32.0,0,0,56.4958,S,Third,man,True,NaN,Southampton,yes,True
334,1,1,female,NaN,1,0,133.6500,S,First,woman,False,NaN,Southampton,yes,False
